### PR TITLE
WIP: Add e2e test for tmpfs

### PR DESCRIPTION
**Pull request by pmorie**
My own stab at implementing an e2e test for tmpfs.

### CODE REVIEW

**Comment by pmorie** [comment]
@jiangyaogao @vmarmol PTAL

File: test/e2e/empty_dir.go [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+			ObjectMeta: api.ObjectMeta{
+				Name: name,
+				Labels: map[string]string{
+					"name": "foo",
+				},
+			},
+			Spec: api.PodSpec{
+				Containers: []api.Container{
+					{
+						Name:  "nginx",
```
Comment (vmarmol) [inline test/e2e/empty_dir.go]: nit: nginx? :)

File: test/e2e/empty_dir.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+								Medium: api.StorageTypeMemory,
+							},
+						},
+					},
+				},
+			},
+		}
+
+		testContainerOutput("tmpfs mount for emptydir", c, pod, []string{
+			"mount type: tmpfs",
```
Comment (vmarmol) [inline test/e2e/empty_dir.go]: Should we also check that it's writeable if we requested R/W?

File: test/e2e/empty_dir.go (hunk 1 above)
Comment (pmorie) [inline test/e2e/empty_dir.go]: @vmarmol This was quick and dirty, totally pilfered from @jiangyaoguo's PR.  If we go with this one, I'll fix it.

**Comment by vmarmol** [comment]
This looks pretty darn good.

**Comment by pmorie** [comment]
@vmarmol per our discussion on IRC, I'm refactoring this to invoke discrete tests based on flags

**Comment by pmorie** [comment]
^ fyi @jiangyaoguo

**Comment by pmorie** [comment]
@vmarmol, @jiangyaoguo and I spoke on irc and decided that he will base his PR on this branch and finish implementing test for the rests of the use cases for ro / rw.  I will close this PR.

**Comment by pmorie** [comment]
Closing; will be pulled into #6166

**Comment by vmarmol** [comment]
Thanks for the heads-up!

**Comment by jiangyaoguo** [comment]
Thanks again.

**Pull request closed without merging**